src/base/tools.lua

```diff
@@ -29,25 +29,34 @@
 --    returns nil.
 ---
 
-	function p.tools.canonical(identifier)
-		local parts
+	function p.tools.normalize(identifier)
 		if identifier:startswith("v") then -- TODO: this should be deprecated?
-			parts = { "msc", identifier }
-		else
-			parts = identifier:explode("-", true, 1)
-
-			-- couple of little hacks here to fix up version names
-			if parts[2] ~= nil then
-				-- 'msc-100' is accepted, but the code expects 'v100'
-				if parts[1] == "msc" and tonumber(parts[2]:sub(1,3)) ~= nil then
-					parts[2] = "v" .. parts[2]
-				end
-
-				-- perform case-correction of the LLVM toolset
-				if parts[2]:startswith("llvm-vs") then
-					parts[2] = "LLVM-" .. parts[2]:sub(6)
-				end
-			end
+			p.warnOnce(identifier, "toolset '%s' is deprecated, toolset 'msc-%s' instead.", identifier, identifier)
```

Comment:
I just put the deprecation message in there, because there is a comment that says "we should deprecate this"...